### ISSUE TITLE
Git tab-completion stopped working for remotes when using `git rebase`

### ISSUE THREAD

**Issue by thepuzzlemaster** [warpdotdev/Warp#5592]
### Dupe Check

- [X] I have [searched Warp bugs](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3ABugs) and there are no duplicates

### Describe the bug

This recently started happening. I can no longer use tab-completion to complete remote names when running git commands in the terminal. I've been using this functionality for years with warp, so this recently started happening.

### To reproduce

enter a folder with an active git repository, which has an `origin` remote
`git rebase or` then press `tab` 
Note that instead of completing `or` into `origin/` it opens a popup with local branches instead.

### Expected behavior

Terminal tab-completes the current line to show:
`git rebase origin/`

### Screenshots

<img width="944" alt="image" src="https://github.com/user-attachments/assets/020a15f2-e184-4f0f-98bd-f9cd393a398e" />

### Operating system

MacOS

### Operating system and version

Sonoma 14.3 (23D56)

### Shell Version

zsh 5.9 (x86_64-apple-darwin23.0)

### Current Warp version

v0.2024.12.10.15.55.stable_04

### Regression

Yes, this bug started recently or with an X Warp version

### Recent working Warp date

Within the last 2 weeks most likely. I use this command pretty regularly and first noticed it yesterday. Although I don't always upgrade Warp right away. 

### Does this block you from using Warp daily?

Yes, this issue prevents me from using Warp daily.

### Is this an issue only in Warp?

Yes, I confirmed that this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by vorporeal** [comment]
I'm noticing that origin branches do show up for `git checkout or`, but not for `git rebase or`.  Is that true for you as well?

**Comment by thepuzzlemaster** [comment]
Yes. I am seeing that as well. Works as before for `get checkout or`

**Comment by thepuzzlemaster** [comment]
Have you actually tried reproducing this issue because it's pretty easy to reproduce using the steps outlined in the initial post? I just upgraded warp to the latest version and confirmed the issue is still definitely happening for me.
